website/docs/components/pagination/partials/code/how-to-use.md

```diff
@@ -1,22 +1,200 @@
+
+Even if the pagination is based on routing, the `onPageChange/onPageSizeChange` callbacks are still available and can be used to respond to the users' actions (eg. for logging, tracking, etc.)
+
+## How to use the "Compact" pagination
+
+By default, the basic use of the pagination provides:
+
+The basic invocation of the "compact" pagination doesn't require any arguments (apart from the event/routing handlers, see below):
+
+```handlebars
+<Hds::Pagination::Compact />
+```
+
+Renders to:
+
+In this variant only the "prev" and "next" navigation controls are displayed.
+
+### Extra arguments
+
+If necessary, it's possible to hide the labels in the controls:
+
+```handlebars
+<Hds::Pagination::Compact @showLabels={{false}} />
+```
+
+### Events handling
+
+The component exposes a callback function that can be used to respond to page changes:
+
+```handlebars
+<Hds::Pagination::Compact
+  @onPageChange={{this.handlePageChange}}
+/>
+```
+
+The `onPageChange` function is invoked when a user interacts with a "prev" or "next" navigation control and so can be used to respond to a "page" change (eg. updating the list of items in the page and/or updating the routing/URL).
+
+### Routing/URL updates
+
+If you want the pagination to change the URL of the page directly (eg. updating the query parameters) you need to pass the routing parameters to the component:
+
+```handlebars
+<Hds::Pagination::Compact
+  @route={{this.demoRouteName}}
+  @queryFunction={{this.demoQueryFunctionCompact}}
+  @isDisabledPrev={{this.demoIsDisabledPrev}}
+  @isDisabledNext={{this.demoIsDisabledNext}}
+/>
+```
+
+where the `@queryFunction` function will be something like this:
+
+```javascript
+get demoQueryFunctionCompact() {
+  return (page) => {
+    return {
+      demoCurrentToken: page === 'prev' ? this.getPrevToken : this.getNextToken,
+      demoExtraParam: 'hello',
+    };
+  };
+}
+```
+
+When the routing parameters are provided, the "prev/next" controls are rendered as links and if the user clicks on one of them the page URL is automatically updated. This means that the component's state is persisted **outside** of the component and so its whole state **must** be "controlled" by the consumer's code (otherwise there would be confliting states).
+
+In this case, the component doesn't update its internal "state" but the value of the state variables is **always** determined by the consumer's controller code via the component's arguments (most of the times, they are directly connected to the query parameters in the URL).
+
+The reason for using a consumer-side function to determine the `query` argument is because it's dynamic (it depends on the value of the `page/cursor` variable) and gives consumers the ability to specify their own query parameters (they will likely be different case by case).
```

Comment:
```suggestion
The reason for using a consumer-side function to determine the `query` argument is because it’s dynamic (it depends on the value of the `page/cursor` variable) and gives consumers the ability to specify their own query parameters (they will likely be different case by case).
```